CMakeLists.txt

```diff
@@ -520,6 +521,10 @@ if(ENABLE_IMPL_COLLECTION)
   add_definitions(-DCV_COLLECT_IMPL_DATA)
 endif()
 
+if (OPENCV_DISABLE_THREAD_SUPPORT)
+  set(WITH_GSTREAMER OFF)
+endif()
```

Comment:
Why do we need this?

It should be handled in `cmake/vars/OPENCV_DISABLE_THREAD_SUPPORT.cmake`: https://github.com/opencv/opencv/pull/19985/files#diff-87b6de92e07d14f041e4b331f8936a5d5a4f2f8e499aa71bc3967de40cff9960

Please avoid any specific changes in root CMakeLists.txt - it should not manage details of 3rdparty components (otherwise we would get unsupportable huge mess)

Reply:
> It should be handled in cmake/vars/OPENCV_DISABLE_THREAD_SUPPORT.cmake
I agree, sorry, I keep forgetting about this file and I end up running in the same issue everytime I was recompiling after changing branch because...

> Ensure that you have clean CMake cache (after code update)
... I was not aware of this trick! Thank you. I am going to remove the last commit  with this change.